design_drafts/components_architecture_and_urdf_examples.md

```diff
@@ -0,0 +1,424 @@
+
+ros2_control implementation examples are presented for the following robot/robot-cell architectures:
+
+1. Industrial Robots with only one interface
+2. Industrial Robots with multiple interfaces (can not be written at the same time)
+3. Robots with multiple interfaces used at the same time
+4. Industrial Robots with a sensor integrated into the robot's control box
+5. Industrial Robots with a sensor connected to ROS computer
+6. Modular Robots with separate communication to each actuator
+7. Modular Robots with actuators not providing states and additional sensors (simple Transmissions)
+8. Modular Robots with separate communication to each "actuator" with multi joints (Transmission Example) - (system component is used)
+9. Sensor only
+10. Actuator Only
+
+
+Note:
+  * Everything within the `<classType>` tag is implemented as a plugin.
+  * The examples below have some `<param>` tags defined for each plugin which are primarily for demonstration, not part of a pre-defined xml schema. Components can define their own parameters.
+#### 1. Industrial Robots with only one interface
+  * the communication is done using proprietary API to communicate with robot control box
+  * Data for all joints is exchanged in batch (at once)
+  * Examples: KUKA RSI
+
+```xml
+  <ros2_control name="2DOF_System_Robot_Position_Only" type="system">
+    <hardware>
+      <classType>ros2_control_demo_hardware/2DOF_System_Hardware_Position_Only</classType>
+      <param name="example_param_write_for_sec">2</param>
+      <param name="example_param_read_for_sec">2</param>
+    </hardware>
+    <joint name="joint1">
+      <classType>ros2_control_components/PositionJoint</classType>
+      <param name="min_values">-1</param>
+      <param name="max_values">1</param>
+    </joint>
+    <joint name="joint2">
+      <classType>ros2_control_components/PositionJoint</classType>
+      <param name="min_values">-1</param>
+      <param name="max_values">1</param>
+    </joint>
+  </ros2_control>
+```
+
+Note:
+  * `ros2_control_components/PositionJoint`type has implicitly:
+    ```xml
+      <commandInterfaceType>position</commandInterfaceType>
+      <stateInterfaceType>position</stateInterfaceType>
+    ```
+
+#### 2. Industrial Robots with multiple interfaces (can not be written at the same time)
+  * the communication is done using proprietary API to communicate with robot control box
+  * Data for all joints is exchanged in batch (at once)
+  * Examples: KUKA FRI, ABB Yummy, Schunk LWA4p, etc.
+
+```xml
+  <ros2_control name="2DOF_System_Robot_MultiInterface" type="system">
+    <hardware>
+      <classType>ros2_control_demo_hardware/2DOF_System_Hardware_MultiInterface</classType>
+      <param name="example_param_write_for_sec">2</param>
+      <param name="example_param_read_for_sec">2</param>
+    </hardware>
+    <joint name="joint1">
+      <classType>ros2_control_components/MultiInterfaceJoint</classType>
+      <commandInterfaceType>position</commandInterfaceType>
```

Comment:
Do we have to make sure at this point that the modes are not mixed up between the joints? Or does it make sense to control joint1 in position and control2 in effort?

Reply:
IMO, we do not have to put limitation on this. I am actually actively using this: on my robot there are six joints, three of them are position-controlled and three velocity-controlled.